src/compiler/codegen/condition_node.cc

```diff
@@ -34,15 +34,55 @@ std::string GetFabsCFunc(std::string const& threshold_type) {
   }
 }
 
+std::string float_to_bin(float number) {
+    std::stringstream ss;
+    ss << "0x" << std::hex << std::setw(8) << std::setfill('0') << *(reinterpret_cast<unsigned int*>(&number));
+    return ss.str();
+}
+
+std::string double_to_bin(double number) {
+    std::stringstream ss;
+    ss << "0x" << std::hex << std::setw(16) << std::setfill('0') << *(reinterpret_cast<unsigned long long*>(&number));
+    return ss.str();
+}
+
+std::string getOppositeOperator(const std::string& op) {
+    if (op == "<") return ">=";
+    else if (op == ">") return "<=";
+    else if (op == "<=") return ">";
+    else if (op == ">=") return "<";
+    else return "";  // unknown operator
+}
+
+std::string thresh_as_int(const std::string& threshold_type, ast::NumericalConditionNode const* node) {
+      std::string negatstring = "";
+      float splitval = std::get<float>(node->threshold_);
+      std::string op = treelite::OperatorToString(node->op_);
+      std::string new_dtype = (threshold_type == "double") ? "long long" : "int";
+
+      if (splitval < 0) {
+          splitval = -splitval;
+          negatstring = " ^ (0b1 << 31)";
+          op = getOppositeOperator(op);
+      }
+
+      std::string split_val_bin = float_to_bin(splitval);
+      return "(*( (("+new_dtype+"*)(data)) + "+std::to_string(node->split_index_)+" )"
+      +negatstring+")"+op+"(("+new_dtype+")("+split_val_bin+"))";
```

Comment:
Let's use `fmt::format` to format strings for the sake of legibility. It works like f-strings in Python.
```suggestion
      return fmt::format("(*( (({new_dtype}*)(data)) + {split_index} ) {negatstring} ) {op} (( {new_dtype} )( {split_val_bin} ))",
          "new_dtype"_a = new_dtype, "split_index"_a = node->split_index_,
          "negatstring"_a = negatstring, "op"_a = op,
          "new_dtype"_a = new_dtype, "split_val_bin"_a = split_val_bin);
```